pkg/manifest/mantaray.go

```diff
@@ -138,22 +138,20 @@ func (m *mantarayManifest) IterateAddresses(ctx context.Context, fn swarm.Addres
 		}
 
 		if node != nil {
-			var stop bool
-
 			if node.Reference() != nil {
 				ref := swarm.NewAddress(node.Reference())
 
-				stop = fn(ref)
-				if stop {
-					return errStopIterator
+				err = fn(ref)
```

Comment:
are these two conditions not mutually exclusive?
if they are you can simplify the code

Reply:
They are not.
First one (`node.Reference()`) contains address to mantaray trie node, and second (`node.Entry()`) for the reference to the actual thing that was stored on that node (if path contains some value, i.e. file).